toucan_connectors/snowflake_common.py

```diff
@@ -142,6 +142,19 @@ def render_datasource(self, datasource: SfDataSource) -> dict:
             'parameters': prepared_query_parameters,
         }
 
+    def add_default_order_if_needed(
+        self,
+        query: str,
+        query_columns: Optional[List] = None,
+    ):
+        """If it's a SELECT type query and there is not an 'ORDER BY' inside, we sort using ROW_NUMBER function"""
+
+        if query_columns and 'WITH SELECT_STEP_' in query and ' ORDER BY ' not in query:
+            # we append the order from row_number on the first column
+            query += f' ORDER BY ROW_NUMBER() OVER (ORDER BY {query_columns[0]} ASC) ASC'
```

Comment:
Is there any advatage of using the `row_number` function instead of just using the first column?
`ORDER BY {query_columns[0]} ASC`

Reply:
Another question about this ordering function: imagine this column does not have unique values, bu contains just a few (like a category). Then this column alone would not be enough to sort the results. To make sure we have always the same order, should we use all columns in the order by clause?

Reply:
For the first question :
It was the first approach in this PR but we decided with @piotch that it should be better to use an ORDER BY ROWNUMBER function on the first column to prevent ordering on a column that have duplicated values !

For the second question : 
(And yes it's related to the First Question)
This is true, a workaround i propose is this : Instead of orther only on one column, order by all the available columns : 
```
SELECT * FROM (
  WITH SELECT_STEP_0 AS (
    SELECT COLUMN1, COLUMN2, COLUMN3 FROM "ROLE2_SCHEMA"."DEFORESTATION_CSV"
  ) SELECT COLUMN1, COLUMN2, COLUMN3 FROM SELECT_STEP_0 
  ORDER BY ROW_NUMBER() OVER (ORDER BY COLUMN1, COLUMN2, COLUMN3) ASC
) LIMIT 10000;
```

What do you think ?

Reply:
OK! Thanks for the explanation!
1. If we do have the insurance that the result of row_number will stay the same across queries, why not! But I couldn't find anything in the docs indicating it, so maybe we would need to test that.

2. Great proposal :+1:

Reply:
updated !